discord/ext/commands/converter.py

```diff
@@ -159,6 +159,80 @@ class UserConverter(IDConverter):
 
         return result
 
+class MessageConverter(IDConverter):
+    """Converts to a :class:`discord.Message`.
+
+    The lookup strategu is as follows (in order):
+
+    1. Lookup by "{channel ID}-{message ID}" (got by shift-clicking on "Copy ID")
+    2. Lookup by message URI
+    3. Lookup by message ID (the message **must** be in the context channel)
+    """
+
+    async def convert(self, ctx, argument: str) -> discord.Message:
```

Comment:
Type hinting is done by the stubs somewhere else.
```suggestion
    async def convert(self, ctx, argument):
```